lib/asciidoctor/reader.rb

```diff
@@ -958,11 +958,11 @@ def preprocess_conditional_directive keyword, target, delimiter, text
         if no_target
           # the text in brackets must match a conditional expression
           if text && EvalExpressionRx =~ text.strip
-            lhs = $1
-            op = $2
-            rhs = $3
+            lhs = resolve_expr_val $1
             # regex enforces a restricted set of math-related operations (==, !=, <=, >=, <, >)
-            skip = ((resolve_expr_val lhs).send op, (resolve_expr_val rhs)) ? false : true
+            op = $2
+            rhs = resolve_expr_val $3
+            skip = (lhs.send op, rhs) ? false : true rescue true
```

Comment:
I was going to say, is it possible to have other errors besides `NoSuchMethod` (such as `ArgumentError`).
`rescue true` will catch any error?

Reply:
I can't think of any other errors at the moment. But regardless, if the operation cannot be performed, it means the condition is not true. So it is correct to skip in this case.